### PR TITLE
Frontend/230/Form Validation

### PR DESCRIPTION
**Pull request by hilvitzs**
### This pull request changes...

- This PR adds form validation to the `EditProfile` page.
- Adds JSDocs to `ComboBox.jsx`, `EditProfile.jsx` (and validation function in EditProfile), `actions/stts.js`, `reducers/stts.js`
- Fix Snyk vulnerability with `react-scripts@3.4.3` - updated to `react-scripts@3.4.4`

**FUNCTIONALITY**
There should be no validation done until the user tries to submit the form. Upon submission, any form field that does not have a valid character (one character for first and last name and a selected STT) will receive a form group error (the line on the side of the form group), a red outline will surround the corresponding input missing its value, and a statement will appear to explain what is missing above the element.

**TO TEST**
- Go to https://tdp-frontend.app.cloud.gov/ and login
- On the login page, focus on the First Name input and then blur away from the input
    - There should be no error yet
- Repeat previous step for Last Name
    - There should be no error yet
- Repeat step for STT
    - There should be no error yet
- Click on the `Request Access` button
- All 3 inputs should have the above mentioned styling changes
- Go to `First Name` input, type in a character and blur out of it and the error messaging should go away
- Repeat for `Last Name`
- Repeat for `STT`

The one outstanding bug is that if you clear your selection in STT and then blur out of the element, it will look like Alabama is selected even though it is not.

### [AC] This pull request is ready to merge when...

- [ ] Tests have been updated (and all tests are passing)
- [ ] Vulnerability Scan Test (Zap) has no high/critical CVEs
- [ ] Linting Tests (ESLint for UI and Flake8 for Python)
- [ ] Unit Tests ( including a code coverage test report)
- [ ] Accessibility Tests (Pa11y UI Only)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screen reader, text scaling) OR an exemption is documented
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

### [DoD] This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by hilvitzs** [comment]
@riatzukiza Tagging you so it's easier to find.

File: tdrs-frontend/docs/security-tracking/sprint6_frontend_10232020.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Sprint 6 Frontend Vulnerability Report (10/23/2020)
+
+This is a report of the vulernabilities discovered during Spring 6, the source of the notification, and the resolution implemented in response to it.
```
Comment (riatzukiza) [inline tdrs-frontend/docs/security-tracking/sprint6_frontend_10232020.md:3]: `Spring 6` is a typo

File: tdrs-frontend/src/components/ComboBox/ComboBox.test.js [hunk 2]
```diff
@@ -0,0 +1,344 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import { render } from '@testing-library/react'
+
+import ComboBox from './ComboBox'
+
+describe('ComboBox', () => {
+  // it('Select element should have value of selected option', () => {
+  //   const props = {
```
Comment (riatzukiza) [inline tdrs-frontend/src/components/ComboBox/ComboBox.test.js:9]: Should we be keeping this commented code, or can we remove it?

**Review by riatzukiza — APPROVED** [review]

File: tdrs-frontend/docs/security-tracking/sprint6_frontend_10232020.md (hunk 1 above)
Comment (hilvitzs) [reply tdrs-frontend/docs/security-tracking/sprint6_frontend_10232020.md:3]: Nice catch

File: tdrs-frontend/src/components/ComboBox/ComboBox.test.js (hunk 2 above)
Comment (hilvitzs) [reply tdrs-frontend/src/components/ComboBox/ComboBox.test.js:9]: Yes, this can be removed. Thank you.

**Pull request merged**